### PR TITLE
[xtro] Implement support for auto-sanitizing the .ignore files by defining an environment variable.

### PR DESCRIPTION
**Pull request by rolfbjarne**
Going through each failure one-by-one after fixing something that fixed many
entries is annoying, so automate the process.

### CODE REVIEW

File: tests/xtro-sharpie/xtro-sanity/Sanitizer.cs [hunk 1]
```diff
@@ -212,12 +221,21 @@ public static int Main (string [] args)
 					continue;
 				var rawfile = Path.ChangeExtension (file, ".raw");
 				var raws = new List<string> (File.ReadAllLines (rawfile));
-				foreach (var entry in File.ReadAllLines (file)) {
+				var failures = new List<string> ();
+				var lines = File.ReadAllLines (file);
```
Comment (mandel-macaque) [inline tests/xtro-sharpie/xtro-sanity/Sanitizer.cs]: We used to get issues when doing ReadAllLines in large files in xharness results. Can we use a reader instead?

Comment (spouliot) [reply tests/xtro-sharpie/xtro-sanity/Sanitizer.cs]: They are not the big - but an easier change is to use `foreach (var line in File.ReadLines(file))` which will read line-by-line (in memory)

**Comment by spouliot** [comment]
@mandel-macaque the `File` class has easier API to do the same. It's also a change that would be needed in several places (in Xtro) and likely better in a different PR

**Comment by mandel-macaque** [comment]
@spouliot fair enough, I just don't want to hit the memory issues we had in xharness, will approve. Should we file an issue?

**Comment by rolfbjarne** [comment]
Yes, these files are quite small, and mostly hand-maintained, so they shouldn't grow so big they cause problems (if they do, we have even bigger problems).

**Comment by spouliot** [comment]
@monojenkins backport to xcode11

**Pull request merged**